packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt

```diff
@@ -698,20 +707,88 @@ class AmplifyDataStorePluginTest {
 
         val nestedSerializedModelOutput = mapOf(
             "id" to "af9cfa64-1ea9-46d6-b9e2-8203179d5392",
-            "name" to "A brilliant Post",
+            "title" to "A brilliant Post",
             "rating" to 5,
             "blog" to SerializedModel.builder().serializedData(
                 mapOf<String, Any>(
                     "name" to "Amazing Blog",
                     "id" to "8cb7d5a5-435d-4632-a890-90ed0c6107f5"
                 ) as HashMap<String, Any>
-            ).modelSchema(null).build()
+            ).modelSchema(blogModelSchema).build()
         )
 
+
         assertEquals(
             nestedSerializedModelOutput,
-            flutterPlugin.deserializeNestedModels(nestedSerializedModelInput)
+            flutterPlugin.deserializeNestedModel(nestedSerializedModelInput, postModelSchema)
+        )
+    }
+
+    @Test
+    fun test_model_nests_custom_type_deserialization() {
```

Comment:
*nit:* I am not sure if you intended to use "nests" or if this was just a typo. I think "nested" makes a little more sense. If you do change it here, a few other test methods in kotlin and swift have the same "nests" naming. 
```suggestion
    fun test_model_nested_custom_type_deserialization() {
```